iredis/client.py

```diff
@@ -36,30 +36,47 @@ class Client:
     iRedis client, hold a redis-py Client to interact with Redis.
     """
 
-    def __init__(self, host, port, db, password=None):
+    def __init__(
+        self,
+        host=None,
+        port=None,
+        db=0,
+        username=None,
+        path=None,
+        password=None,
+        scheme="redis",
+    ):
         self.host = host
         self.port = port
+        self.path = path
         self.db = db
-        if config.decode:
-            self.connection = Connection(
-                host=self.host,
-                port=self.port,
-                db=self.db,
-                password=password,
-                encoding=config.decode,
-                decode_responses=True,
-                encoding_errors="replace",
-                socket_keepalive=config.socket_keepalive,
-            )
+        self.username = username
+        connection_class = Connection
+        connection_kwargs = {
+            "db": db,
+            "password": password,
+            "socket_keepalive": config.socket_keepalive,
+        }
+        if scheme in ("redis", "rediss"):
+            connection_kwargs["host"] = host
```

Comment:
create a new `connection_kwargs` instead of updating the common one would be better